src/new/wallet.js

```diff
@@ -2565,6 +2565,109 @@ class HathorWallet extends EventEmitter {
 
     return graphvizData;
   }
+
+  /**
+   * This function is responsible for getting the details of each token in the transaction.
+   * @param {Object} wallet the current wallet
+   * @param {{ [tokenUid]: number }} tokenBalances Object with the token uid as key and the balance as value
+   * @returns {Promise<Array<{ txId: string, timestamp: number, version: number, voided: boolean, weight: number, tokenName: string, tokenSymbol: string, balance: number }>> } Array of token details
+   * @example
+   * [
+   *  {
+   *   txId: '000021e7addbb94a8e43d7f1237d556d47efc4d34800c5923ed3a75bf5a2886e';
+   *   timestamp: 123456789;
+   *   version: 1;
+   *   voided: false;
+   *   weight: 18.5;
+   *   tokenId: '00',
+   *   tokenName: 'Hathor',
+   *   tokenSymbol: 'HTR',
+   *   balance: 100,
+   *  },
+   * ]
+   */
+  async getTxById(txId) {
+    /**
+     * This function is responsible for calculate the balance of each token in the transaction.
+     * that belongs to the wallet.
+     * @param {{ txId: string, timestamp: number, voided: boolean }} tx
+     * @returns {{ [tokenUid]: number }} Object with the token uid as key and the balance as value
+     */
+    const getTokensBalance = (tx) => {
+      const mineOutputs = tx.outputs.filter((output) => this.isAddressMine(output.decoded.address));
+      const tokensBalance = mineOutputs.reduce((acc, output) => {
+        const { token, value } = output;
+        if (acc[token]) {
```

Comment:
Maybe do something like

```typescript
if (token in acc) {
```

So this does not gets evaluated as `false` if `acc[token]` is 0

Reply:
I removed this function.